usaspending_api/awards/views.py

```diff
@@ -6,15 +6,81 @@
 from usaspending_api.common.views import DetailViewSet, AutocompleteView
 from usaspending_api.common.serializers import AggregateSerializer
+from usaspending_api.common.exceptions import InvalidParameterException
+from django.db.models import F, Sum
+from django.db import Error
 
 AggregateItem = namedtuple('AggregateItem', ['field', 'func'])
 
 
+class AwardAggregateViewSet(SuperLoggingMixin,
+                            FilterQuerysetMixin,
+                            AggregateQuerysetMixin,
+                            DetailViewSet):
+
+    serializer_class = AggregateSerializer
+
+    """Return aggregated award information."""
+    def get_queryset(self):
+        queryset = Award.objects.all()
+        queryset = self.filter_records(self.request, queryset=queryset)
+        queryset = self.aggregate(self.request, queryset=queryset)
+        queryset = self.order_records(self.request, queryset=queryset)
+        return queryset
+
+
+class AwardAutocomplete(FilterQuerysetMixin,
+                        AutocompleteView):
+    """Autocomplete support for award summary objects."""
+    # Maybe refactor this out into a nifty autocomplete abstract class we can just inherit?
+    serializer_class = AwardSerializer
+
+    def get_queryset(self):
+        """Return the view's queryset."""
+        queryset = Award.nonempty.all()
+        queryset = self.serializer_class.setup_eager_loading(queryset)
+        filtered_queryset = self.filter_records(self.request, queryset=queryset)
+        ordered_queryset = self.order_records(self.request, queryset=filtered_queryset)
+        return ordered_queryset
+
+
+class AwardTypeAwardSpendingViewSet(DetailViewSet):
+    """Return all award spending by award type for a given fiscal year and agency id"""
+
+    serializer_class = AwardTypeAwardSpendingSerializer
+
+    def get_queryset(self):
+        # retrieve post request payload
+        json_request = self.request.query_params
+
+        # retrieve fiscal_year & awarding_agency_id from request
+        fiscal_year = json_request.get('fiscal_year', None)
+        awarding_agency_id = json_request.get('awarding_agency_id', None)
+
+        # required query parameters were not provided
+        if not (fiscal_year and awarding_agency_id):
+            raise InvalidParameterException('Missing one or more required query parameters: fiscal_year, awarding_agency_id')
+
+        try:
+            queryset = Transaction.objects.all()
+            # Filter based on fiscal year and agency id
+            queryset = queryset.filter(fiscal_year=fiscal_year, awarding_agency=awarding_agency_id)
+            # alias awards.category to be award_type
+            queryset = queryset.annotate(award_type=F('award__category'))
+            # sum obligations for each category type
+            queryset = queryset.values('award_type').annotate(obligated_amount=Sum('federal_action_obligation'))
+        except Error:
```

Comment:
I'm not sure the reasoning of this catch - it just catches a specific exception and re-raises it as a vague exception, destroying information that could be useful for debugging; unless I'm missing something, that doesn't seem good.  Could we catch it with `except Error as e` and add `str(e)` to the exception message?